tests/test_2d_operations.cpp

```diff
@@ -1170,3 +1173,300 @@ TEST_CASE("swap_edge", "[operations][swap][2D]")
+{
+    using namespace operations;
+    SECTION("split_single_triangle")
+    {
+        //         0
+        //        / \  
+        //       2   1
+        //      /  0  \  
+        //     /       \ 
+        //  1  ----0---- 2
+        //
+        // this case covered the on boundary case
+        DEBUG_TriMesh m = single_triangle();
+        Tuple f = m.edge_tuple_between_v1_v2(1, 2, 0);
+        spdlog::info("{}", m.id(f, PV));
+        spdlog::info("{}", m.id(m.switch_vertex(f), PV));
+        spdlog::info("{}", m.id(m.switch_vertex(m.switch_edge(f)), PV));
+        OperationSettings<tri_mesh::FaceSplit> settings;
+        settings.initialize_invariants(m);
+        wmtk::operations::tri_mesh::FaceSplit face_split_op(m, f, settings);
+        bool is_success = face_split_op();
+        Tuple ret = face_split_op.return_tuple();
+        spdlog::info("{}", m.id(ret, PV));
+        spdlog::info("{}", m.id(m.switch_vertex(ret), PV));
+        spdlog::info("{}", m.id(m.switch_vertex(m.switch_edge(ret)), PV));
+        REQUIRE(is_success);
+        REQUIRE(m.get_all(PV).size() == 4);
+        REQUIRE(!m.is_boundary_vertex(ret));
+        REQUIRE(!m.is_boundary_edge(ret));
+        REQUIRE(!m.is_boundary_edge(m.switch_edge(ret)));
+        REQUIRE(m.id(ret, PV) == 4);
+        REQUIRE(m.id(m.switch_vertex(ret), PV) == 1);
+        REQUIRE(m.id(m.switch_vertex(m.switch_edge(ret)), PV) == 2);
+        REQUIRE(SimplicialComplex::vertex_one_ring(m, ret).size() == 3);
+    }
+    SECTION("split in quad")
+    {
+        //  3--1--- 0
+        //   |     / \ .
+        //   2 f1 /2   1
+        //   |  0/ f0  \ .
+        //   |  /       \ .
+        //  1  ----0---- 2
+        //
+        DEBUG_TriMesh m = interior_edge();
+        Tuple f = m.edge_tuple_between_v1_v2(1, 0, 1);
+        spdlog::info("{}", m.id(f, PV));
+        spdlog::info("{}", m.id(m.switch_vertex(f), PV));
+        spdlog::info("{}", m.id(m.switch_vertex(m.switch_edge(f)), PV));
+        Tuple split_ret;
+        {
+            OperationSettings<tri_mesh::EdgeSplit> op_settings;
+            op_settings.initialize_invariants(m);
+            tri_mesh::EdgeSplit split_op(m, f, op_settings);
+            split_op();
+            split_ret = split_op.return_tuple();
+        }
+        spdlog::info("{}", m.id(split_ret, PV));
+        spdlog::info("{}", m.id(m.switch_vertex(split_ret), PV));
+        spdlog::info("{}", m.id(m.switch_vertex(m.switch_edge(split_ret)), PV));
+        const Tuple coll_input_tuple = m.switch_face(m.switch_edge(split_ret));
+        OperationSettings<tri_mesh::EdgeCollapse> collapse_settings;
+        collapse_settings.initialize_invariants(m);
+        tri_mesh::EdgeCollapse coll_op(m, coll_input_tuple, collapse_settings);
+        coll_op();
+        const Tuple& coll_ret = coll_op.return_tuple();
+        auto ret = m.switch_vertex(m.switch_edge(coll_ret));
+        spdlog::info("{} {}", m.id(ret, PV), m.id(m.switch_vertex(ret), PV));
```

Comment:
I guess that is still debug code but `CHECK`s are missing and the tests should not contain any debug messages, if not necessary.